Refactoring Tests for User model

lirantal opened the pull request:
re-factoring the server model tests to create user model from schema only during tests, so that we can re-use mongodb's _v versioning fields, and also cleaning up user tests on each test iteration

ilanbiala commented on modules/users/tests/server/user.server.model.tests.js:
```diff
@@ -57,59 +57,90 @@ describe('User Model Unit Tests:', function () {
+            should.not.exist(err);
+            done();
+          });
+          
```
nitpick, but inconsistent space? If we want more spacing, we should have above and below.

ilanbiala commented on modules/users/tests/server/user.server.model.tests.js:
```diff
@@ -57,59 +57,90 @@ describe('User Model Unit Tests:', function () {
-        var passwordAfter = user.password;
-        passwordBefore.should.equal(passwordAfter);
-        done();
+
```
Spacing

ilanbiala commented on modules/users/tests/server/user.server.model.tests.js:
```diff
@@ -57,59 +57,90 @@ describe('User Model Unit Tests:', function () {
 
     it('should be able to save 2 different users', function (done) {
-      user.remove(function (err) {
+
```
spacing

ilanbiala commented:
Looks good, just some minor nitpicks.

lirantal commented:
Thanks @ilanbiala , missed your CR :)
I addressed the spacing but I updated the convention there to have a line space between var declares and the rest of the code for readability as otherwise it just looks a lot of code mangled together.

ilanbiala commented:
That looks awesome to me!

lirantal commented:
Cool, thanks.

codydaig commented:
LGTM
